src/clj/game/cards/operations.clj

```diff
@@ -77,7 +77,7 @@
                                      :effect (effect (clear-wait-prompt :corp)
                                                      (game.core/run eid serv nil card))}
                        :no-ability {:effect (effect (clear-wait-prompt :corp)
-                                                    (as-agenda :corp (last (:discard corp)) 1))
+                                                    (as-agenda :corp (first (filter #(= (:cid card) (:cid %)) (:discard corp))) 1))
```

Comment:
A slightly better way to do this pattern (finding the first instance of something matching a filter predicate):

`(some #(when (= (:cid card) (:cid %)) %) (:discard corp))`

`some` takes a function and a collection, and returns the first first truthy value found when applying the function to each element of the collection in turn. Here, the `when` will test your condition and return a card if the card fits the condition, otherwise nil; the `some` will continue iterating cards until it finds a card that the function returns non-nil with, i.e., a card that matches the filter.